src/app/bot.ts

```diff
@@ -69,6 +71,30 @@ export class Bot {
         }
       });
     });
+
+    // load slash plugins
+    const slashPluginFolder = path.join(__dirname, '/slash_plugins');
+    fs.readdir(slashPluginFolder, (_err, files) => {
+      files.forEach(async (file) => {
+        if (!file.endsWith('.ts') && !file.endsWith('.js')) {
+          return;
+        }
+
+        const plugin = await import(`./slash_plugins/${file}`).then((m) => m.default);
+
+        const result = SlashCommand.safeParse(plugin);
+        // FIXME we could validate more here: for example, checking the parameters
+        // property, but this should be fine for now
+        if (result.success) {
+          slashCommands.set(plugin.commandName, result.data as ISlashCommand);
+        } else {
+          this.container.loggerService.warn(
+            `${file} does not \`export default\` a plugin with type ISlashCommand!`
+          );
+          this.container.loggerService.warn(result.error);
+        }
+      });
+    });
```

Comment:
This needs to be done somewhere else, or change the code flow of the instantiation of the `Listener`

The `Listener` is called in the [constructor](https://github.com/cs-discord-at-ucf/lion/blob/8ade0f3a100be08e4edf81acff8c484ee2e81b99/src/app/bot.ts#L25) of `Bot`
Which is before `_loadAndRun` is called by the `run` function

Listener is created [here](https://github.com/cs-discord-at-ucf/lion/blob/8ade0f3a100be08e4edf81acff8c484ee2e81b99/src/index.ts#L7) which tries to loop over the (currently empty) `slashCommands` map
`_loadAndRun` is called [here](https://github.com/cs-discord-at-ucf/lion/blob/8ade0f3a100be08e4edf81acff8c484ee2e81b99/src/index.ts#L8C1-L8C1) which populates the `slashCommands` map

Reply:
Hmm, where would you prefer? The code works in its current form. I bypassed several important-sounding classes with `Handler` and `Listener` in them because I just couldn't understand what they were for.

Reply:
I thought this would be the right place because this is where the regular plugins get set up.

Reply:
the latter would be easier, if possible

Reply:
This is odd, because I just tested this and the upload appears to work as intended. I deleted 8ball (the only slash command rn) and reran and there was nothing. Then, I added 8ball back and reran and it was there at `/8ball`

Reply:
I wonder if it's a permissions thing? You need to [grant the slash commands permission](https://discordjs.guide/preparations/adding-your-bot-to-servers.html#creating-and-using-your-invite-link) to the bot in your server

Reply:
My reproduction:
```
git switch suneettipirneni/feature/slash-commands
npm ci
npm run watch
```